src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -27,15 +27,33 @@
 ;; ## Plugin Initialization
 
 (defn initialize!
-  ; 1. Access app info using app accessor to determine url, app-id, etc.
-  ; 2. Create a vault client
-  ; 3. Authenticate the vault client
-  ; 4. Return the vault client
   "Set up the vault client."
   [logger app-accessor]
   (alter-var-root #'log/logger (constantly logger))
-  ;; TODO: actually initialize a Vault client
-  nil)
+  (atom nil))
+
+
+;; ## Model
+
+;; A map of user configurable fields to the all the data necessary to define those fields
+(def input-schema
+  {;; the name of the corresponding input field
+   :vault_addr {;; the metadata required by the plugin API about every input field
```

Comment:
presumably these are underscore for some javascript compatibility reason? Otherwise clojure ❤️ kebab-case.

Reply:
Yep, js stuff for all of those. kebab-case is best case 😋